src/python/WMCore/MicroService/Unified/Transferor.py

```diff
@@ -72,88 +86,290 @@ def close(self):
+        except ValueError as err: # store is closed
+            self.logger.error(err)
+            requests = []
+	except Exception as err: # general error
+            self.logger.error(err)
+            requests = []
+        # perform check of requests
+        for request in requests:
+            # Once all transfer requests were successfully made, update: assigned -> staging
+            status = self.checkTransferRequests(request)
+            if status:
+                self.logger.debug("### all transfer requests were successfull")
+                self.change(request, 'staging', '### transferor')
+            # if there is nothing to be transferred (no input at all), then update: staging -> staged
+            # uncomment when we'll deal with real transfers
+            #self.checkStatus(request)
 
     def stop(self):
-        "Stop RequestManager"
+        "Stop MSManager"
+        # stop MSTransferorMonit thread
+        self.ms_monit.stop()
+        # stop MSTransferor thread
         self.thr.stop() # stop checkStatus thread
         status = self.thr.running()
         return status
 
-    def checkPhedex(self, request):
+    def transferRequest(self, request, rdict):
+        "Send request to Phedex and return status of request subscription"
+	datasets = rdict.get('datasets', [])
+	sites = rdict.get('sites', [])
+        if datasets and sites:
+	    subscription = PhEDExSubscription(datasets, sites, self.group)
+	    self.logger.debug("### add subscription %s for request %s" % (subscription, request))
+            # TODO: implement how to get transfer id
+            tid = hashlib.md5(str(subscription)).hexdigest()
+	    # TODO: when ready enable submit subscription step
+	    # self.phedex.subscribe(subscription)
+            return tid
+
+    def getTransferIDsDoc(self):
+        """
+        Get transfer ids document from backend. The document has the following form:
+        https://gist.github.com/amaltaro/72599f995b37a6e33566f3c749143154
+	{"wf_A": {"timestamp": 0000
+		  "primary": {"dset_1": ["list of transfer ids"]},
+		  "secondary": {"PU_dset_1": ["list of transfer ids"]},
+	 "wf_B": {"timestamp": 0000
+		  "primary": {"dset_1": ["list of transfer ids"],
+			      "parent_dset_1": ["list of transfer ids"]},
+		  "secondary": {"PU_dset_1": ["list of transfer ids"],
+				"PU_dset_2": ["list of transfer ids"]},
+         ...
+	}
+        """
+        doc = {}
+        return doc
+
+    def updateTransferIDs(self, trecords):
+        "Update transfer ids in backend"
+        # TODO/Wait: https://github.com/dmwm/WMCore/issues/9198
+        doc = self.getTransferIDsDoc()
+        pass
+
+    def getTransferIDs(self):
+        "Get transfer ids from backend"
+        # TODO/Wait: https://github.com/dmwm/WMCore/issues/9198
+        # meanwhile return transfer ids from internal store 
+        tids = [r['tid'] for r in self.store.records()]
+        return tids
+
+    def getRequest(self, tid):
```

Comment:
It should be the other way around, return transfer id for a given request (sort of as implemented above).

Reply:
ok, I remove this method (i used it for testing purposes).